kstreams/types.py

```diff
@@ -24,3 +24,7 @@ def __call__(
         serializer: typing.Optional["Serializer"] = None,
         serializer_kwargs: typing.Optional[typing.Dict] = None,
     ) -> typing.Awaitable[RecordMetadata]: ...
+
+
+D = typing.TypeVar("D")
```

Comment:
Do we need the `generic`? It seems that it is always used to annotated `typing.Optional[Deserializer]`

Reply:
Yes, because we can use the `Deprecated` type to deprecate any attribute, I think it's a handy type to have